TypeCobol/Compiler/AntlrUtils/ParserDiagnosticErrorListener.cs

```diff
@@ -57,18 +57,18 @@ public override void SyntaxError(IRecognizer recognizer, IToken offendingSymbol,
     public class ParserDiagnostic : Diagnostic
     {
 		public ParserDiagnostic(string message, IToken offendingSymbol, string ruleStack, MessageCode code = MessageCode.SyntaxErrorInParser, Exception exception = null) :
-			base(code, offendingSymbol == null ? -1 : offendingSymbol.Column, offendingSymbol == null ? -1 : (offendingSymbol.StopIndex < 0 ? -1 : (offendingSymbol.StopIndex+1)), offendingSymbol == null ? -1 : offendingSymbol.Line, message, exception)
+			base(code, offendingSymbol == null ? -1 : offendingSymbol.Column, offendingSymbol == null ? -1 : (offendingSymbol.StopIndex < 0 ? -1 : (offendingSymbol.StopIndex+1)), offendingSymbol == null ? -1 : offendingSymbol.Line, message, exception, offendingSymbol)
 		{
 			OffendingSymbol = offendingSymbol;
 			this.ruleStack = ruleStack;
-
-            // TO DO - IMPORTANT : this is the INITIAL line number, and not the CURRENT line number
-            // This is enough to pass all unit tests, but will return false informations in real usage !
-            // for real line number, use a Snapshot
-            if (Line < 0 && OffendingSymbol != null)
+            var copyDirective = (OffendingSymbol as ImportedToken)?.CopyDirective;
+            if (copyDirective != null)
             {
-                CodeElement e = OffendingSymbol as CodeElement;
-                if (e != null && e.ConsumedTokens.Count > 0) Line = e.ConsumedTokens[0].Line;
+                Line = copyDirective.COPYToken.Line;
+                bool endsWithDot = Message.EndsWith(".");
+                if (endsWithDot) Message = Message.Substring(0, Message.Length - 1);
+                Message += $", in copy '{copyDirective.TextName}'";
+                if (endsWithDot) Message += ".";
```

Comment:
I think it is slightly better to have indication of the copy at the beginning of the message so the developer can quickly determine whether the error is in his own file or another. Somehing like : `Error in copy <copy-name> : <original message>`. This also simplifies the code : no need to care about dots.

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/pull/1602/commits/7c04857a0a5144ffc90df51a3c52657caa9fab40